### PR TITLE
TSK-1239: Assertj use in every module

### PR DESCRIPTION
**Pull request by tge20**
Update of Spring Boot version to 2.4.0
Thus and because it is a rule in our guidelines I exchanged all remaining Assertion (junit) uses with the Assertj framework. Moreover Alphanumeric is deprecated since that version, thus change to MethodName.

---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [x] Commit message format → TSK-XXX: Your commit message.
- [x] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [x] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [x] PR fulfills the ticket
- [x] Edge cases and unwanted side effects are tested
- [x] Readability

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,354 +1,354 @@
 <project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (mustaphazorgati) [inline pom.xml:1]: what did you do here? please reset this file

Comment (mustaphazorgati) [reply pom.xml:1]: `git checkout upstream/master pom.xml`

File: rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/TestSchemaNameCustomizable.java [hunk 2]
```diff
@@ -44,7 +43,7 @@ void resetDb() throws SQLException {
   @Test
   void checkCustomSchemaNameIsDefined_Postgres() throws SQLException {
     resetDb();
-    Assume.assumeTrue("Test only executed with Postgres database", isPostgres);
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/TestSchemaNameCustomizable.java:47]: This is not an assertions. This is an assumption.
Please use `Assumptions.assumeThat(isPostgres).isTrue();` (with a static import)

File: rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/TestSchemaNameCustomizable.java [hunk 3]
```diff
@@ -66,7 +65,7 @@ void checkCustomSchemaNameIsDefined_Postgres() throws SQLException {
   @Test
   void checkCustomSchemaNameIsDefined_OtherDb() throws SQLException {
     resetDb();
-    Assume.assumeFalse("Test only executed if NOT Postgres", isPostgres);
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/TestSchemaNameCustomizable.java:69]: This is not an assertions. This is an assumption.
Please use `Assumptions.assumeThat(isPostgres).isFalse();` (with a static import)

File: rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/TestSchemaNameCustomizable.java [hunk 4]
```diff
@@ -8,7 +8,6 @@
 import java.sql.SQLException;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/TestSchemaNameCustomizable.java:8]: @gitgoodjhe what the hell is this class? Can we please talk about this?

- resetDb() is not annotated with `@BeforeEach`
- class name does not end with `Test` and thus this class will not be found by surefire.
- general setup is questionable
- test content is too complicated.. 

Like what the hell?!

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

File: rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/TestSchemaNameCustomizable.java (hunk 4 above)
Comment (gitgoodjhe) [reply rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/TestSchemaNameCustomizable.java:8]: First time I ever saw this class :)

**Comment by tge20** [comment]
https://sonarcloud.io/dashboard?branch=TSK-1239&id=tge20_taskana

File: pom.xml (hunk 1 above)
Comment (tge20) [reply pom.xml:1]: I just updated the Spring Boot version, nothing else....

**Comment by mustaphazorgati** [comment]
@tge20 you forgot to change the spring version to 2.4.0. That's why the build is red. Otherwise this PR looks fine.

**Comment by mustaphazorgati** [comment]
please prefix the commits wiht `TSK-1239:`

**Comment by tge20** [comment]
> 
> 
> @tge20 you forgot to change the spring version to 2.4.0. That's why the build is red. Otherwise this PR looks fine.

**Comment by tge20** [comment]
> 
> 
> @tge20 you forgot to change the spring version to 2.4.0. That's why the build is red. Otherwise this PR looks fine.

I know, because above you said I should reset the file...

**Comment by gitgoodjhe** [comment]
you need to change mybatis from 2.0.5 to 2.0.6 to resolve the merge conflict :)

**Review by mustaphazorgati — APPROVED** [review]

**Pull request merged**